In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
!unrar e /content/drive/'My Drive'/DataMining/dataset.rar


UNRAR 5.50 freeware      Copyright (c) 1993-2017 Alexander Roshal


Extracting from /content/drive/My Drive/DataMining/dataset.rar


Would you like to replace the existing file product.xlsx
13561766 bytes, modified on 2019-03-14 19:45
with a new one
13561766 bytes, modified on 2019-03-14 19:45

[Y]es, [N]o, [A]ll, n[E]ver, [R]ename, [Q]uit Q

Program aborted


In [ ]:
import pandas as pd 
import numpy as np
import datetime
import re
print('done')

done


\\
# **Orders Dataset**

In [ ]:
df_orders = pd.read_csv('orders.csv')
df_orders

ID_Order  ID_Customer  ...  city_name_fa Quantity_item
0        2714054       469662  ...    محمود آباد           1.0
1       11104039      3063877  ...        خرمدره           1.0
2        4228130      3184893  ...          قرچک           1.0
3       22225624      6888562  ...            قم           1.0
4        4068771      2533490  ...         تهران           1.0
...          ...          ...  ...           ...           ...
199995  12172771      4403268  ...         تهران           2.0
199996   4425393      3745774  ...        محمدیه           1.0
199997   6671889      4845514  ...        اصفهان           1.0
199998   6315995      2541418  ...         همدان           1.0
199999   7584127      1342585  ...    بندر انزلی           1.0

[200000 rows x 7 columns]

\\
## Method 1: 

$$ completeness = \frac {records\ with\ no\ missing\ attribute}{Total\ records\ in\ a\ dataset}$$

In [ ]:
noNan_df = df_orders.dropna(axis=0)
noNan_df

ID_Order  ID_Customer  ...  city_name_fa Quantity_item
0        2714054       469662  ...    محمود آباد           1.0
1       11104039      3063877  ...        خرمدره           1.0
2        4228130      3184893  ...          قرچک           1.0
3       22225624      6888562  ...            قم           1.0
4        4068771      2533490  ...         تهران           1.0
...          ...          ...  ...           ...           ...
199995  12172771      4403268  ...         تهران           2.0
199996   4425393      3745774  ...        محمدیه           1.0
199997   6671889      4845514  ...        اصفهان           1.0
199998   6315995      2541418  ...         همدان           1.0
199999   7584127      1342585  ...    بندر انزلی           1.0

[200000 rows x 7 columns]

In [ ]:
completeness_percent = len(noNan_df) / len(df_orders) * 100
print(f"completeness with method 1 : {completeness_percent} %" )

completeness with method 1 : 100.0 %


## Method 2

$$ completeness = \frac {A: number\ of\ data\ required\ for\ the\ particular\ context\ in\ the\ data file}{B: number\ of\ data\ in\ the\ specified\ particular\ context\ of\ intended\ use}$$

We have selected 2 more important columns and stored them in new dataframe.

In [ ]:
new_df_orders = df_orders[['Amount_Gross_Order', 'Quantity_item']]
new_df_orders

Amount_Gross_Order  Quantity_item
0                 597982.0            1.0
1                 980000.0            1.0
2                 229358.0            1.0
3                  16514.0            1.0
4                 133028.0            1.0
...                    ...            ...
199995            344037.0            2.0
199996            600000.0            1.0
199997            450000.0            1.0
199998            193486.0            1.0
199999             55046.0            1.0

[200000 rows x 2 columns]

delete rows which have at least 1 Nan.

In [ ]:
noNan_new_df = new_df_orders.dropna(axis=0)
noNan_new_df

Amount_Gross_Order  Quantity_item
0                 597982.0            1.0
1                 980000.0            1.0
2                 229358.0            1.0
3                  16514.0            1.0
4                 133028.0            1.0
...                    ...            ...
199995            344037.0            2.0
199996            600000.0            1.0
199997            450000.0            1.0
199998            193486.0            1.0
199999             55046.0            1.0

[200000 rows x 2 columns]

In [ ]:
completeness_percent = len(noNan_new_df) / len(df_orders) * 100
print(f"completeness with method 2 :{completeness_percent} %" )

completeness with method 2 :100.0 %


## Method 3

$$ completeness = \frac {A: attribute\ fields\ containing\ values}{B: records\ \times attributes}$$

In [ ]:
number_of_notnulls = df_orders.notnull().sum().sum()
print("Total number of null cells: ", number_of_notnulls)

Total number of null cells:  1400000


In [ ]:
completeness_percent = number_of_notnulls / (df_orders.shape[0] * df_orders.shape[1]) * 100
print(f"completeness with method 3 :{completeness_percent} %" )

completeness with method 3 :100.0 %


\\
# **comment Dataset**

In [ ]:
df_comment = pd.read_excel('comment.xlsx')
df_comment

product_id  ...                                            comment
0            90213  ...  سلام، قبل اینکه نظرم رو بگم میخواستم به یک موض...
1            56871  ...  بوی تند ولی خوشبو داره.ماندگاریش خوبه و هر چقد...
2            49738  ...  متاسفانه عمر مفید این ماشین کم هست و بعد از دو...
3           134367  ...  افا از وقتی من اینو واسه پسرم خریدم همه بچه ها...
4           151402  ...  این تیغ محصول کشور آمریکا هست ( در عکس محصول م...
...            ...  ...                                                ...
100003      104998  ...  قیمت مناسبی داره در حال حاضر . با این پول سخت ...
100004      333389  ...  در کل راضیم درسته لنوو با قیمت پایین تر هسته و...
100005      356546  ...  من این محصول رو از بیرون تهیه کردم . و باید بگ...
100006       33874  ...  دوسال و.نیم پیش از دیجی کالا گرفتم. هنوز مثل ر...
100007      318278  ...  با استفاده از این شامپو مشکل خشکی مو و پوست سر...

[100008 rows x 3 columns]

\\
## Method 1: 

$$ completeness = \frac {records\ with\ no\ missing\ attribute}{Total\ records\ in\ a\ dataset}$$

In [ ]:
noNan_df = df_comment.dropna(axis=0)
noNan_df

product_id  ...                                            comment
0            90213  ...  سلام، قبل اینکه نظرم رو بگم میخواستم به یک موض...
1            56871  ...  بوی تند ولی خوشبو داره.ماندگاریش خوبه و هر چقد...
2            49738  ...  متاسفانه عمر مفید این ماشین کم هست و بعد از دو...
3           134367  ...  افا از وقتی من اینو واسه پسرم خریدم همه بچه ها...
4           151402  ...  این تیغ محصول کشور آمریکا هست ( در عکس محصول م...
...            ...  ...                                                ...
100003      104998  ...  قیمت مناسبی داره در حال حاضر . با این پول سخت ...
100004      333389  ...  در کل راضیم درسته لنوو با قیمت پایین تر هسته و...
100005      356546  ...  من این محصول رو از بیرون تهیه کردم . و باید بگ...
100006       33874  ...  دوسال و.نیم پیش از دیجی کالا گرفتم. هنوز مثل ر...
100007      318278  ...  با استفاده از این شامپو مشکل خشکی مو و پوست سر...

[99903 rows x 3 columns]

In [ ]:
completeness_percent = len(noNan_df) / len(df_comment) * 100
print(f"completeness with method 1 : {completeness_percent} %" )

completeness with method 1 : 99.89500839932805 %


## Method 2

$$ completeness = \frac {A: number\ of\ data\ required\ for\ the\ particular\ context\ in\ the\ data file}{B: number\ of\ data\ in\ the\ specified\ particular\ context\ of\ intended\ use}$$

We have selected 2 more important columns and stored them in new dataframe.

In [ ]:
new_df_comment = df_comment[['confirmed_at', 'comment']]
new_df_comment

confirmed_at                                            comment
0       2018-08-23 11:05:47  سلام، قبل اینکه نظرم رو بگم میخواستم به یک موض...
1       2018-08-23 10:21:53  بوی تند ولی خوشبو داره.ماندگاریش خوبه و هر چقد...
2       2018-10-13 11:13:39  متاسفانه عمر مفید این ماشین کم هست و بعد از دو...
3       2018-10-17 11:27:57  افا از وقتی من اینو واسه پسرم خریدم همه بچه ها...
4       2018-08-23 10:28:47  این تیغ محصول کشور آمریکا هست ( در عکس محصول م...
...                     ...                                                ...
100003  2018-07-09 21:27:12  قیمت مناسبی داره در حال حاضر . با این پول سخت ...
100004  2018-07-09 21:26:52  در کل راضیم درسته لنوو با قیمت پایین تر هسته و...
100005  2018-07-09 21:26:30  من این محصول رو از بیرون تهیه کردم . و باید بگ...
100006  2018-07-09 21:26:12  دوسال و.نیم پیش از دیجی کالا گرفتم. هنوز مثل ر...
100007  2018-07-09 21:26:02  با استفاده از این شامپو مشکل خشکی مو و پوست سر...

[100008 rows x 2 columns]

delete rows which have at least 1 Nan.

In [ ]:
noNan_new_df = new_df_comment.dropna(axis=0)
noNan_new_df

confirmed_at                                            comment
0       2018-08-23 11:05:47  سلام، قبل اینکه نظرم رو بگم میخواستم به یک موض...
1       2018-08-23 10:21:53  بوی تند ولی خوشبو داره.ماندگاریش خوبه و هر چقد...
2       2018-10-13 11:13:39  متاسفانه عمر مفید این ماشین کم هست و بعد از دو...
3       2018-10-17 11:27:57  افا از وقتی من اینو واسه پسرم خریدم همه بچه ها...
4       2018-08-23 10:28:47  این تیغ محصول کشور آمریکا هست ( در عکس محصول م...
...                     ...                                                ...
100003  2018-07-09 21:27:12  قیمت مناسبی داره در حال حاضر . با این پول سخت ...
100004  2018-07-09 21:26:52  در کل راضیم درسته لنوو با قیمت پایین تر هسته و...
100005  2018-07-09 21:26:30  من این محصول رو از بیرون تهیه کردم . و باید بگ...
100006  2018-07-09 21:26:12  دوسال و.نیم پیش از دیجی کالا گرفتم. هنوز مثل ر...
100007  2018-07-09 21:26:02  با استفاده از این شامپو مشکل خشکی مو و پوست سر...

[99903 rows x 2 columns]

In [ ]:
completeness_percent = len(noNan_new_df) / len(df_comment) * 100
print(f"completeness with method 2 :{completeness_percent} %" )

completeness with method 2 :99.89500839932805 %


## Method 3

$$ completeness = \frac {A: attribute\ fields\ containing\ values}{B: records\ \times attributes}$$

In [ ]:
number_of_notnulls = df_comment.notnull().sum().sum()
print("Total number of null cells: ", number_of_notnulls)

Total number of null cells:  299919


In [ ]:
completeness_percent = number_of_notnulls / (df_comment.shape[0] * df_comment.shape[1]) * 100
print(f"completeness with method 3 :{completeness_percent} %" )

completeness with method 3 :99.96500279977602 %


The results are:


1.   Method 1 : 99.89500839932805 %
2.   Method 2 : 99.89500839932805 %
3.   Method 3 : 99.96500279977602 %

This shows that the first two methods give equal percentages of completeness, but method 3 the percentage more than the two other methods.



\\
# **keifiat Dataset**

In [ ]:
df_keifiat = pd.read_excel('keifiat.xlsx')
df_keifiat

product_id  ...                                      disadvantages
0            3692  ...                                                NaN
1           90213  ...                                          ["ندارد"]
2           59473  ...                                                NaN
3          120499  ...  ["کیفیت پایین اسپیکر\r","حاشیه خالی زیر صفحه ن...
4           67200  ...                                    ["ظرفیت متوسط"]
...           ...  ...                                                ...
99995      549845  ...                                          ["ندارد"]
99996      784160  ...                 ["محل عبور بند کفش ها ایراد دارد"]
99997      820985  ...                                          ["ندیدم"]
99998      678073  ...                                                NaN
99999      451791  ...                                                NaN

[100000 rows x 12 columns]

\\
Firstly we should replace the '\N' values with np.nan

In [ ]:
df_keifiat = df_keifiat.replace(to_replace = '\\N', value=np.nan)
df_keifiat

product_id  ...                                      disadvantages
0            3692  ...                                                NaN
1           90213  ...                                          ["ندارد"]
2           59473  ...                                                NaN
3          120499  ...  ["کیفیت پایین اسپیکر\r","حاشیه خالی زیر صفحه ن...
4           67200  ...                                    ["ظرفیت متوسط"]
...           ...  ...                                                ...
99995      549845  ...                                          ["ندارد"]
99996      784160  ...                 ["محل عبور بند کفش ها ایراد دارد"]
99997      820985  ...                                          ["ندیدم"]
99998      678073  ...                                                NaN
99999      451791  ...                                                NaN

[100000 rows x 12 columns]

\\
## Method 1: 

$$ completeness = \frac {records\ with\ no\ missing\ attribute}{Total\ records\ in\ a\ dataset}$$

In [ ]:
noNan_df = df_keifiat.dropna(axis=0)
noNan_df

product_id  ...                                      disadvantages
1           90213  ...                                          ["ندارد"]
3          120499  ...  ["کیفیت پایین اسپیکر\r","حاشیه خالی زیر صفحه ن...
4           67200  ...                                    ["ظرفیت متوسط"]
5          133722  ...  ["تاچ پدش میتونست بهتر باشه از نظر سرعت پاسخ گ...
6          148509  ...                                          ["ندارد"]
...           ...  ...                                                ...
99968      590683  ...                         ["رنگ رفتن \r","خراب شدن"]
99970      256517  ...                      ["در آمدن و احتماس شکستن مچ"]
99983       87150  ...  ["پایین بودن ظرفیت واقعی در حدود ۴۰۰۰\r","زمان...
99995      549845  ...                                          ["ندارد"]
99997      820985  ...                                          ["ندیدم"]

[18813 rows x 12 columns]

In [ ]:
completeness_percent = len(noNan_df) / len(df_keifiat) * 100
print(f"completeness with method 1 : {completeness_percent} %" )

completeness with method 1 : 18.813 %


## Method 2

$$ completeness = \frac {A: number\ of\ data\ required\ for\ the\ particular\ context\ in\ the\ data file}{B: number\ of\ data\ in\ the\ specified\ particular\ context\ of\ intended\ use}$$

We have selected 2 more important columns and stored them in new dataframe.

In [ ]:
new_df_keifiat = df_keifiat[['likes', 'dislikes', 'verification_status', 'recommend']]
new_df_keifiat

likes  dislikes verification_status        recommend
0          0         0            verified              NaN
1          4         1            verified      recommended
2          1         0            verified  not_recommended
3          6        11            verified          no_idea
4         19         4            verified          no_idea
...      ...       ...                 ...              ...
99995      0         0            verified      recommended
99996      7        32            verified              NaN
99997      0         0            verified      recommended
99998      2         1            verified          no_idea
99999      0         0            verified          no_idea

[100000 rows x 4 columns]

delete rows which have at least 1 Nan.

In [ ]:
noNan_new_df = new_df_keifiat.dropna(axis=0)
noNan_new_df

likes  dislikes verification_status        recommend
1          4         1            verified      recommended
2          1         0            verified  not_recommended
3          6        11            verified          no_idea
4         19         4            verified          no_idea
5          6         1            verified      recommended
...      ...       ...                 ...              ...
99994      0         0            verified          no_idea
99995      0         0            verified      recommended
99997      0         0            verified      recommended
99998      2         1            verified          no_idea
99999      0         0            verified          no_idea

[63618 rows x 4 columns]

In [ ]:
completeness_percent = len(noNan_new_df) / len(df_keifiat) * 100
print(f"completeness with method 2 :{completeness_percent} %" )

completeness with method 2 :63.617999999999995 %


## Method 3

$$ completeness = \frac {A: attribute\ fields\ containing\ values}{B: records\ \times attributes}$$

In [ ]:
number_of_notnulls = df_keifiat.notnull().sum().sum()
print("Total number of null cells: ", number_of_notnulls)

Total number of null cells:  1036585


In [ ]:
completeness_percent = number_of_notnulls / (df_keifiat.shape[0] * df_keifiat.shape[1]) * 100
print(f"completeness with method 3 :{completeness_percent} %" )

completeness with method 3 :86.38208333333334 %


The results are:


1.   Method 1 : 18.813 %
2.   Method 2 : 63.617999999999995 %
3.   Method 3 : 86.38208333333334 %

This shows that the method returned a very short perdentage, but the two other methods returned bigger perndentages and agein the method 3 shows the most perdentage for the completeness of the dataframe.



\\
# **product Dataset**

In [ ]:
df_product = pd.read_excel('product.xlsx')
df_product

id  ...                                 product_attributes
0      723302  ...  [{"Key":"وزن بسته‌بندی","Value":"480"},{"Key":...
1      727089  ...  [{"Key":"دستگیره","Value":"1 جفت"},{"Key":"نوع...
2      727096  ...  [{"Key":"حجم","Value":"50"},{"Key":"نوع عصاره"...
3      727803  ...  [{"Key":"ابعاد رو بالشی","Value":"50×70"},{"Ke...
4      725222  ...  [{"Key":"در","Value":"ندارد"},{"Key":"جنس","Va...
...       ...  ...                                                ...
99995  803375  ...                                                NaN
99996  790932  ...                                                NaN
99997  786604  ...                                                NaN
99998  799047  ...                                                NaN
99999  782276  ...                                                NaN

[100000 rows x 10 columns]

\\
Firstly we should replace the '\N' values with np.nan

In [ ]:
df_product = df_product.replace(to_replace = '\\N', value=np.nan)
df_product

id  ...                                 product_attributes
0      723302  ...  [{"Key":"وزن بسته‌بندی","Value":"480"},{"Key":...
1      727089  ...  [{"Key":"دستگیره","Value":"1 جفت"},{"Key":"نوع...
2      727096  ...  [{"Key":"حجم","Value":"50"},{"Key":"نوع عصاره"...
3      727803  ...  [{"Key":"ابعاد رو بالشی","Value":"50×70"},{"Ke...
4      725222  ...  [{"Key":"در","Value":"ندارد"},{"Key":"جنس","Va...
...       ...  ...                                                ...
99995  803375  ...                                                NaN
99996  790932  ...                                                NaN
99997  786604  ...                                                NaN
99998  799047  ...                                                NaN
99999  782276  ...                                                NaN

[100000 rows x 10 columns]

\\
## Method 1: 

$$ completeness = \frac {records\ with\ no\ missing\ attribute}{Total\ records\ in\ a\ dataset}$$

In [ ]:
noNan_df = df_product.dropna(axis=0)
noNan_df

id  ...                                 product_attributes
3     727803  ...  [{"Key":"ابعاد رو بالشی","Value":"50×70"},{"Ke...
8     725806  ...  [{"Key":"پازل","Value":"1000 تکه"},{"Key":"وزن...
9     725822  ...  [{"Key":"ابعاد","Value":"80*20*20"},{"Key":"مش...
10    724904  ...  [{"Key":"کشور مبدا برند"},{"Key":"آستین","Valu...
14    723227  ...  [{"Key":"نوع عصاره","Value":"-"},{"Key":"شماره...
...      ...  ...                                                ...
3012  725667  ...  [{"Key":"سایر توضیحات","Value":"مناسب برای نوج...
3014  723174  ...  [{"Key":"نوع","Value":"آویز"},{"Key":"ریموت","...
3018  727083  ...  [{"Key":"ابعاد","Value":"10.5 × 3.5"},{"Key":"...
3054  726846  ...  [{"Key":"کشور مبداء برند","Value":"ایران"},{"K...
3103  725534  ...  [{"Key":"سایر توضیحات","Value":""},{"Key":"لیو...

[522 rows x 10 columns]

In [ ]:
completeness_percent = len(noNan_df) / len(df_product) * 100
print(f"completeness with method 1 : {completeness_percent} %" )

completeness with method 1 : 0.522 %


## Method 2

$$ completeness = \frac {A: number\ of\ data\ required\ for\ the\ particular\ context\ in\ the\ data file}{B: number\ of\ data\ in\ the\ specified\ particular\ context\ of\ intended\ use}$$

We have selected 2 more important columns and stored them in new dataframe.

In [ ]:
new_df_product = df_product[['category_title_fa', 'brand_name_fa']]
new_df_product

category_title_fa brand_name_fa
0          ماگ، لیوان و فنجان        لومانا
1                آبکش و آبگیر         تیتیز
2                   روغن صورت          کدلی
3                  سرویس خواب          هوبی
4          ماگ، لیوان و فنجان        متفرقه
...                       ...           ...
99995                   سوتین        متفرقه
99996   محافظ صفحه نمایش گوشی        متفرقه
99997              تابلو شاسی        متفرقه
99998                    سینی       برایتون
99999  تی‌شرت و پولوشرت زنانه        تارپون

[100000 rows x 2 columns]

delete rows which have at least 1 Nan.

In [ ]:
noNan_new_df = new_df_product.dropna(axis=0)
noNan_new_df

category_title_fa brand_name_fa
0          ماگ، لیوان و فنجان        لومانا
1                آبکش و آبگیر         تیتیز
2                   روغن صورت          کدلی
3                  سرویس خواب          هوبی
4          ماگ، لیوان و فنجان        متفرقه
...                       ...           ...
99995                   سوتین        متفرقه
99996   محافظ صفحه نمایش گوشی        متفرقه
99997              تابلو شاسی        متفرقه
99998                    سینی       برایتون
99999  تی‌شرت و پولوشرت زنانه        تارپون

[100000 rows x 2 columns]

In [ ]:
completeness_percent = len(noNan_new_df) / len(df_product) * 100
print(f"completeness with method 2 :{completeness_percent} %" )

completeness with method 2 :100.0 %


## Method 3

$$ completeness = \frac {A: attribute\ fields\ containing\ values}{B: records\ \times attributes}$$

In [ ]:
number_of_notnulls = df_product.notnull().sum().sum()
print("Total number of null cells: ", number_of_notnulls)

Total number of null cells:  801370


In [ ]:
completeness_percent = number_of_notnulls / (df_product.shape[0] * df_product.shape[1]) * 100
print(f"completeness with method 3 :{completeness_percent} %" )

completeness with method 3 :80.137 %


The results are:


1.   Method 1 : 0.522 %
2.   Method 2 : 100 %
3.   Method 3 : 80.137 %

This shows that the method 1 returned o.522 % which is a very short number. Insted of the method 1, method 2 return 100 percent in which we selected our important features so the results shows, we have no missing value for our important features. 
The method 3 also returned 80.137 & which is not a short number but it is less than the method 2.


\\
# **tarikhche kharid Dataset**

In [ ]:
df_tarikhche = pd.read_csv('tarikhche kharid.csv')
df_tarikhche

id  product_variant_id  ...  product_id  marketplace_seller_id
0        18752043             1977385  ...      667969                   2962
1        18752022             1977380  ...      150144                  78525
2        18752046             1977380  ...      150144                  78525
3        18751982             1977374  ...      908541                   4199
4        18752018             1977374  ...      908541                   4199
...           ...                 ...  ...         ...                    ...
2184353  12763117             1155042  ...      221109                   7664
2184354  13703581             1155042  ...      221109                   7664
2184355  13703591             1155042  ...      221109                   7664
2184356  13703595             1155042  ...      221109                   7664
2184357  16766304             1155042  ...      221109                   7664

[2184358 rows x 15 columns]

\\ 
As can be seen in the dataframe, the columns base_price, buy_price, start_at, end_at and tags are empty so we should drop the from the dataframe.

In [ ]:
df_tarikhche = df_tarikhche.drop(['base_price', 'buy_price', 'start_at', 'end_at', 'tags'], axis = 1)
df_tarikhche

id  product_variant_id  ...  product_id  marketplace_seller_id
0        18752043             1977385  ...      667969                   2962
1        18752022             1977380  ...      150144                  78525
2        18752046             1977380  ...      150144                  78525
3        18751982             1977374  ...      908541                   4199
4        18752018             1977374  ...      908541                   4199
...           ...                 ...  ...         ...                    ...
2184353  12763117             1155042  ...      221109                   7664
2184354  13703581             1155042  ...      221109                   7664
2184355  13703591             1155042  ...      221109                   7664
2184356  13703595             1155042  ...      221109                   7664
2184357  16766304             1155042  ...      221109                   7664

[2184358 rows x 10 columns]

\\
Firstly we should replace the '\N' values with np.nan

In [ ]:
df_tarikhche = df_tarikhche.replace(to_replace = '\\N', value=np.nan)

\\
## Method 1: 

$$ completeness = \frac {records\ with\ no\ missing\ attribute}{Total\ records\ in\ a\ dataset}$$

In [ ]:
noNan_df = df_tarikhche.dropna(axis=0)
noNan_df

id  product_variant_id  ...  product_id  marketplace_seller_id
0        18752043             1977385  ...      667969                   2962
1        18752022             1977380  ...      150144                  78525
2        18752046             1977380  ...      150144                  78525
3        18751982             1977374  ...      908541                   4199
4        18752018             1977374  ...      908541                   4199
...           ...                 ...  ...         ...                    ...
2184353  12763117             1155042  ...      221109                   7664
2184354  13703581             1155042  ...      221109                   7664
2184355  13703591             1155042  ...      221109                   7664
2184356  13703595             1155042  ...      221109                   7664
2184357  16766304             1155042  ...      221109                   7664

[2184358 rows x 10 columns]

In [ ]:
completeness_percent = len(noNan_df) / len(df_tarikhche) * 100
print(f"completeness with method 1 : {completeness_percent} %" )

completeness with method 1 : 100.0 %


## Method 2

$$ completeness = \frac {A: number\ of\ data\ required\ for\ the\ particular\ context\ in\ the\ data file}{B: number\ of\ data\ in\ the\ specified\ particular\ context\ of\ intended\ use}$$

We have selected 2 more important columns and stored them in new dataframe.

In [ ]:
new_df_tarikhche = df_tarikhche[['selling_price', 'rrp_price', 'active', 'created_at', 'order_limit', 'show_in_price_history']]
new_df_tarikhche

selling_price  rrp_price  ...  order_limit show_in_price_history
0               850000     850000  ...            1                     1
1             12550000   12550000  ...            2                     1
2             12500000   12500000  ...            2                     1
3               310000     310000  ...            2                     1
4               300000     300000  ...            2                     1
...                ...        ...  ...          ...                   ...
2184353         439000     529000  ...            2                     1
2184354         469000     529000  ...            2                     1
2184355         465000     529000  ...            2                     1
2184356         460000     460000  ...            2                     1
2184357         449000     449000  ...            2                     1

[2184358 rows x 6 columns]

delete rows which have at least 1 Nan.

In [ ]:
noNan_new_df = new_df_tarikhche.dropna(axis=0)
noNan_new_df

selling_price  rrp_price  ...  order_limit show_in_price_history
0               850000     850000  ...            1                     1
1             12550000   12550000  ...            2                     1
2             12500000   12500000  ...            2                     1
3               310000     310000  ...            2                     1
4               300000     300000  ...            2                     1
...                ...        ...  ...          ...                   ...
2184353         439000     529000  ...            2                     1
2184354         469000     529000  ...            2                     1
2184355         465000     529000  ...            2                     1
2184356         460000     460000  ...            2                     1
2184357         449000     449000  ...            2                     1

[2184358 rows x 6 columns]

In [ ]:
completeness_percent = len(noNan_new_df) / len(df_tarikhche) * 100
print(f"completeness with method 2 :{completeness_percent} %" )

completeness with method 2 :100.0 %


## Method 3

$$ completeness = \frac {A: attribute\ fields\ containing\ values}{B: records\ \times attributes}$$

In [ ]:
number_of_notnulls = df_tarikhche.notnull().sum().sum()
print("Total number of null cells: ", number_of_notnulls)

Total number of null cells:  21843580


In [ ]:
completeness_percent = number_of_notnulls / (df_tarikhche.shape[0] * df_tarikhche.shape[1]) * 100
print(f"completeness with method 3 :{completeness_percent} %" )

completeness with method 3 :100.0 %


The results are:


1.   Method 1 : 100 %
2.   Method 2 : 100 %
3.   Method 3 : 100 %

As you can see in these results, the completeness in all the methods was 100 % which shows that there are 0 cells with NaN value in our new dataframe. (We had dropped the columns which had a lots of NaN value.
